deepchecks/base/check_context.py

```diff
@@ -0,0 +1,279 @@
+                 model_name: str = '',
+                 features_importance: pd.Series = None,
+                 feature_importance_force_permutation: bool = False,
+                 feature_importance_timeout: int = None,
+                 scorers: Mapping[str, Union[str, Callable]] = None,
+                 non_avg_scorers: Mapping[str, Union[str, Callable]] = None
+                 ):
+        # Validations
+        if train is None and test is None and model is None:
+            raise DeepchecksValueError('At least one dataset (or model) must be passed to the method!')
+        if train is not None:
+            train = Dataset.ensure_not_empty_dataset(train)
+        if test is not None:
+            test = Dataset.ensure_not_empty_dataset(test)
+        # If both dataset, validate they fit each other
+        if train and test:
+            if not Dataset.datasets_share_label(train, test):
+                raise DatasetValidationError('train and test requires to have and to share the same label')
+            if not Dataset.datasets_share_features(train, test):
+                raise DatasetValidationError('train and test requires to share the same features columns')
+            if not Dataset.datasets_share_categorical_features(train, test):
+                raise DatasetValidationError(
+                    'train and test datasets should share '
+                    'the same categorical features. Possible reason is that some columns were'
+                    'inferred incorrectly as categorical features. To fix this, manually edit the '
+                    'categorical features using Dataset(cat_features=<list_of_features>'
+                )
+            if not Dataset.datasets_share_index(train, test):
+                raise DatasetValidationError('train and test requires to share the same index column')
+            if not Dataset.datasets_share_date(train, test):
+                raise DatasetValidationError('train and test requires to share the same date column')
+        if test and not train:
+            raise DatasetValidationError('Can\'t initialize context with only test. if you have single dataset, '
+                                         'initialize it as train')
+        if model is not None:
+            # Here validate only type of model, later validating it can predict on the data if needed
+            model_type_validation(model)
+
+        self._train = train
+        self._test = test
+        self._model = model
+        self._feature_importance_force_permutation = feature_importance_force_permutation
+        self._features_importance = features_importance
+        self._feature_importance_timeout = feature_importance_timeout
+        self._calculated_importance = False
+        self._importance_type = None
+        self._validated_model = False
+        self._task_type = None
+        self._user_scorers = scorers
+        self._user_non_avg_scorers = non_avg_scorers
```

Comment:
do we have a better name for this? e.g. per_class_scorers

Reply:
Possible will change in another PR